### ISSUE TITLE
`unnecessary_self_imports` doesn't trigger in nested imports

### ISSUE THREAD

**Issue by ada4a** [rust-lang/rust-clippy#17652]
### Summary

The lint does lint on `bar::{self}` in
```rs
use foo::bar::{self};
```
but not in
```rs
use foo::{
    bar::{self},
    /* other imports, optionally */
};
```

Seen in https://codeberg.org/mergiraf/mergiraf/pulls/813#issuecomment-21940430

### Lint Name

unnecessary_self_imports

### Reproducer

I tried this code:

```rs
mod foo { pub mod bar {} }

use foo::{
    bar::{self},
};
```

I expected to see this happen:

```
warning: import ending with `::{self}`
 --> src/lib.rs:4:5
  |
4 |     bar::{self},
  |     -----------
  |     |
  |     help: consider omitting `::{self}`: `bar`
  |
```

Instead, this happened:
(no warning)

### Version

```text
rustc 1.98.0 (88d9e12ae 2026-08-18)
binary: rustc
commit-hash: 88d9e12ae178fab0fb5cc050a94da85685d449ea
commit-date: 2026-08-18
host: x86_64-unknown-linux-gnu
release: 1.98.0
LLVM version: 22.1.8
```
